From f5767ddce5d49ce8daecb5a9325c5df1789e6788 Mon Sep 17 00:00:00 2001
From: Wim Taymans <wtaymans@redhat.com>
Date: Wed, 26 Feb 2020 12:22:12 +0100
Subject: [PATCH 1/2] alsa: disable UCM when a profile_set is set in udev

When we configure a PROFILE_SET property in udev, prefer to use that
one over the UCM profile.
---
 src/modules/alsa/module-alsa-card.c | 12 ++++++++----
 1 file changed, 8 insertions(+), 4 deletions(-)

diff --git a/src/modules/alsa/module-alsa-card.c b/src/modules/alsa/module-alsa-card.c
index 1e1090fe0..2100ff9c7 100644
--- a/src/modules/alsa/module-alsa-card.c
+++ b/src/modules/alsa/module-alsa-card.c
@@ -803,6 +803,13 @@ int pa__init(pa_module *m) {
         goto fail;
     }
 
+    /* If a profile set was set with udev, use that and don't use ucm. */
+#ifdef HAVE_UDEV
+    fn = pa_udev_get_property(u->alsa_card_index, "PULSE_PROFILE_SET");
+    if (fn != NULL)
+	    u->use_ucm = false;
+#endif
+
     /* Force ALSA to reread its configuration. This matters if our device
      * was hot-plugged after ALSA has already read its configuration - see
      * https://bugs.freedesktop.org/show_bug.cgi?id=54029
@@ -831,9 +838,6 @@ int pa__init(pa_module *m) {
     }
     else {
         u->use_ucm = false;
-#ifdef HAVE_UDEV
-        fn = pa_udev_get_property(u->alsa_card_index, "PULSE_PROFILE_SET");
-#endif
 
         if (pa_modargs_get_value(u->modargs, "profile_set", NULL)) {
             pa_xfree(fn);
@@ -841,8 +845,8 @@ int pa__init(pa_module *m) {
         }
 
         u->profile_set = pa_alsa_profile_set_new(fn, &u->core->default_channel_map);
-        pa_xfree(fn);
     }
+    pa_xfree(fn);
 
     if (!u->profile_set)
         goto fail;
-- 
2.24.1

